pkg/kubenest/controller/global.node.controller/global_node_lifecycle_controller_test.go

```diff
@@ -0,0 +1,348 @@
+// nolint
+package globalnodecontroller
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/kosmos.io/kosmos/pkg/apis/kosmos/v1alpha1"
+)
+
+func GetGlobalNode(globalnodelist []*v1alpha1.GlobalNode, name string) (*v1alpha1.GlobalNode, error) {
+	// 模拟查找返回节点
+	for _, node := range globalnodelist {
+		if node.Name == name { // 假设 GlobalNode 结构体中有 Name 字段
+			return node, nil
+		}
+	}
+	return nil, fmt.Errorf("GlobalNode not found")
+}
+
+func UpdateStatus(globalnodelist []*v1alpha1.GlobalNode, globalNode *v1alpha1.GlobalNode) error {
+	// 遍历 globalnodelist 查找目标节点
+	var nodeFound bool
+	for i, node := range globalnodelist {
+		if node.Name == globalNode.Name {
+			// 找到匹配的节点，更新该节点
+			globalnodelist[i].Status.Conditions = globalNode.Status.Conditions
+			nodeFound = true
+			break // 找到目标节点后退出循环
+		}
+	}
+
+	// 如果没有找到对应的节点，返回错误
+	if !nodeFound {
+		return fmt.Errorf("GlobalNode with name '%s' not found", globalNode.Name)
+	}
+
+	// 返回成功，表示更新完成
```

Comment:
Remove unnecessary comments; simple comments can be replaced with the function name directly.